config/registry/default_schema.yaml

```diff
@@ -0,0 +1,693 @@
+      constraints:
+      - !ForeignKeyConstraint
+        src: skymap
+        tgt: SkyMap.skymap
+      - !ForeignKeyConstraint
+        src:
+        - skymap
+        - tract
+        tgt:
+        - Tract.skymap
+        - Tract.tract
+    
+    - !DataUnit
+      name: Label
+      doc: >
+        An arbitrary string value.  There is no SQL representation or constraint on
+        the values a Label can take.
+      columns:
+      - !Column &label
+        name: label
+        type: string
+  
+  tables:
+    - !Table
+      name: Datasets
+      columns:
+      - !Column
+        name: dataset_id
+        type: int
+        primary_key: true
+        nullable: false
+        doc: >
+          Autoincrement
+      - !Column
+        name: registry_id
+        type: int
+        primary_key: true
+        nullable: false
+        doc: >
+          A different number for each dataset_id incrementer (usually each different
+          Registry).
+      - !Column
+        name: dataset_type_name
+        type: string
+        nullable: false
+      - !Column
+        name: uri
+        type: string
+      - !Column
+        name: run_id
+        type: int
+        nullable: false
+        doc: >
+          Link to the Run table.
+      - !Column
+        name: quantum_id
+        type: int
+        doc: >
+          Link to the Quantum table (only present for SuperTask-generated Datasets).
+      # DataUnit values
+      - !Column
+        <<: *camera
+        nullable: true
+      - !Column
+        <<: *abstract_filter
+        nullable: true
+      - !Column
+        <<: *physical_filter
+        nullable: true
+      - !Column
+        <<: *sensor
+        nullable: true
+      - !Column
+        <<: *visit
+        nullable: true
+      - !Column
+        <<: *exposure
+        nullable: true
+      - !Column
+        <<: *valid_first
+        nullable: true
+      - !Column
+        <<: *valid_last
+        nullable: true
+      - !Column
+        <<: *skymap
+        nullable: true
+      - !Column
+        <<: *tract
+        nullable: true
+      - !Column
+        <<: *patch
+        nullable: true
+      - !Column
+        <<: *skypix
+        nullable: true
+      - !Column
+        <<: *label
+        nullable: true
+      constraints:
+      - !ForeignKeyConstraint
+        src: dataset_type_name
+        tgt: DatasetType.dataset_type_name
+      - !ForeignKeyConstraint
+        src:
+        - run_id
+        - registry_id
+        tgt:
+        - Run.run_id
+        - Run.registry_id
+      - !ForeignKeyConstraint
+        src:
+        - producer_id
+        - registry_id
+        tgt:
+        - Quantum.quantum_id
+        - Quantum.registry_id
+  
+    - !Table
+      name: DatasetComposition
+      doc: >
+        Self-join table that relates components of a dataset to their parent.
+      columns:
+      - !Column
+        name: parent_dataset_id
+        type: int
+        primary_key: true
+        nullable: false
```

Comment:
IN SQL, `primary_key=True` always implies `nullable=False`.  Should we take advantage of that?

Reply:
Probably, although I'd want to check the standard on that to be sure (e.g. the actual primary key in `sqllite` is some auto-generated thing that you don't see so perhaps it does allow it to be `nullable`? Not sure.